rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,225 @@
+
+Once this feed is implemented, RTR MUST adopt it as its source of truth for determining trip assignments in the data that it publishes. If RTR receives a `vehicle_trip_assignment` event for a trip ID that it does not recognize (via either the GTFS schedule or via a trip add `trips_updated` event), it SHOULD map that to a unique added trip ID of its own and maintain that mapping in the GTFS-rt feed for the duration of the trip (that is, the unrecognized trip ID from Glides should not suddenly become mapped to a different randomly-generated added trip). This is to handle the case where RTR is aware of a different set of trip IDs due to a disruption modeled in `gtfs_creator` but not in HASTUS. See [Rationale and alternatives](#rationale-and-alternatives) for some brief discussion of the tradeoff involved here.
+
+### Required data for Glides to produce trip assignments
+[required-rtr-data]: #required-rtr-data
+
+Having Glides handle all trip assignments will require Glides having a direct feed of all vehicle position data. Currently, RTR excludes some known non-revenue yard pull-outs and pull-backs, which makes sense from a passenger information perspective. Internal consumers like Glides will need a more direct feed of the data before any filtering is applied, in addition to certain non-public data fields like the train's AVI code. While the exact mechanism for achieving this is out of scope for this RFC, it is worth noting as a requirement as it relates to other discussions around re-architecting RTR. One possibility that has been discussed is splitting RTR into a data aggregation and state tracking component and a predictions component, set up such that other applications like Glides can also be consumers of the data from the former.
+
+For determining train location for the purposes of performing trip matching, existing data from RTR in the form of direction ID and next stop ID for trains will suffice for the time being.
```

Comment:
Didn't we say that AVI would be required, too?

Reply:
It is - that's mentioned on line 160.